lighthouse-core/audits/dobetterweb/uses-http2.js

```diff
@@ -43,67 +58,174 @@ class UsesHTTP2Audit extends Audit {
+   *
+   * @param {Array<{url: string}>} results
+   * @param {Node} graph
+   * @param {Simulator} simulator
+   * @return {number}
+   */
+  static computeWasteWithTTIGraph(results, graph, simulator) {
+    const beforeLabel = `uses-http2-before`;
+    const afterLabel = `uses-http2-after`;
+    const flexibleOrdering = true;
+
+    const urlsToChange = new Set(results.map(result => result.url));
+    const simulationBefore = simulator.simulate(graph, {label: beforeLabel, flexibleOrdering});
+
+    // Update all the protocols to reflect implementing our recommendations
+    /** @type {Map<string, string>} */
+    const originalProtocols = new Map();
+    graph.traverse(node => {
+      if (node.type !== 'network') return;
+      if (!urlsToChange.has(node.record.url)) return;
+
+      originalProtocols.set(node.record.requestId, node.record.protocol);
+      node.record.protocol = 'h2';
+    });
+
+    const simulationAfter = simulator.simulate(graph, {label: afterLabel, flexibleOrdering});
+
+    // Restore the original protocol after we've done our simulation
+    graph.traverse(node => {
+      if (node.type !== 'network') return;
+      const originalProtocol = originalProtocols.get(node.record.requestId);
+      if (originalProtocol === undefined) return;
+      node.record.protocol = originalProtocol;
+    });
+
+    const savingsOnOverallLoad = simulationBefore.timeInMs - simulationAfter.timeInMs;
+    const savingsOnTTI = Interactive.getLastLongTaskEndTime(simulationBefore.nodeTimings) -
+      Interactive.getLastLongTaskEndTime(simulationAfter.nodeTimings);
+    const savings = Math.max(savingsOnTTI, savingsOnOverallLoad);
+
+    // Round waste to nearest 10ms
+    return Math.round(Math.max(savings, 0) / 10) * 10;
+  }
+
+  /**
+   * Determines whether a network request is a "static resource" that would benefit from H2 multiplexing.
+   * XHRs, tracking pixels, etc generally don't benefit as much because they aren't requested en-masse
+   * for the same origin at the exact same time.
+   *
+   * @param {LH.Artifacts.NetworkRequest} networkRequest
+   * @return {boolean}
+   */
+  static isStaticAsset(networkRequest) {
+    if (!STATIC_RESOURCE_TYPES.has(networkRequest.resourceType)) return false;
+
+    // Resources from third-parties that are less than 100 bytes are usually tracking pixels, not actual resources.
+    // They can masquerade as static types though (gifs, documents, etc)
+    if (networkRequest.resourceSize < 100) {
+      const entity = ThirdPartyAudit.getEntitySafe(networkRequest.url);
+      if (entity) return false;
+    }
+
+    return true;
+  }
+
+  /**
+   * Determine the set of resources that aren't HTTP/2 but should be.
+   * We're a little conservative about what we surface for a few reasons:
+   *
+   *    - The simulator approximation of HTTP/2 is a little more generous than reality.
+   *    - There's a bit of debate surrounding HTTP/2 due to its worse performance in environments with high packet loss.
```

Comment:
This is news to me. Resources to learn more?

Reply:
tl;dr - you're pinning all your hopes on a single connection instead of hedging your bets with 6

https://news.ycombinator.com/item?id=19086639
https://www.twilio.com/blog/2017/10/http2-issues.html
https://www.cachefly.com/http-2-is-not-a-magic-bullet/

will add some of these to the comment 👍